### ISSUE TITLE
Misleading stack trace when spying

### ISSUE THREAD

**Issue by TWiStErRob** [mockito/mockito#1490]
Mockito Version: 2.22.0
Java: 1.7 (compiled with Android Studio / Gradle)
Android Studio 3.1.4
Build #AI-173.4907809, built on July 23, 2018
JRE: 1.8.0_152-release-1024-b02 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

Originally the issue was discovered with Kotlin, https://github.com/nhaarman/mockito-kotlin and Inline MockMaker in the mix, but even after removing all those from the SSCCE, the issue still persisted.

Consider the following production code:
```java
interface Collaborator {
    String getSomething();
}

class SUT {
    private final Collaborator collaborator;

    SUT(Collaborator collaborator) {
        this.collaborator = collaborator;
    }
    void delegating(String arg) {
        delegatee(arg);
    }
    void delegatee(String arg) { // private on this fixes the issue
        stubbed(arg + "x"); // NPE stack trace shows this
    }
    void stubbed(String toBeCaptured) {
        //noinspection RedundantStringToString,ResultOfMethodCallIgnored
        collaborator
                .getSomething()
                .toString() // NPE happens on this member access
        ;
    }
}
```
and assume I want to test that calling `delegating` ends up in `stubbed` with a transformed argument, but I don't want to test what's inside `stubbed` (hence `spy`).
Let's imagine that I made a mistake while writing this test as outlined in the comments: I forgot to add the `doNothing()` stub, which means that the production code of the method I don't want to call is called.

Here's how I would translate this to a JUnit Mockito test:
```java
import org.junit.Assert;
import org.junit.Test;
import org.mockito.ArgumentCaptor;
import org.mockito.Mockito;

public class SUTTest {
    @Test public void testDelegatingFlowsDataToStubbed() {
        SUT sut = Mockito.spy(new SUT(Mockito.mock(Collaborator.class))); // removing spy fixes the issue
        // suppose I forgot to add
        // Mockito.doNothing().when(sut).stubbed(Mockito.<String>any());
        // if doNothing() was there, there would be no need to stub the collaborator
        // but because doNothing() is missing it'll interact with the collaborator and
        // it'll return the default null from getSomething() and crash (this works as expected)

        sut.delegating("hello"); // calling delegatee directly fixes the issue

        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
        Mockito.verify(sut).stubbed(captor.capture());
        Assert.assertEquals("hellox", captor.getValue());
    }
}
```
Note: for a quick repro you can drop all the above code (prod + test) into the same file.

Adding the `doNothing()` will make the test pass, but without the test errors. This error is the subject of this report as it is misleading and makes it hard to fix the test/production code after getting the error.

When running the above code I get the following stack trace:
```ruby
java.lang.NullPointerException
	at x.SUT.delegatee(SUTTest.java:22)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at x.SUT.delegatee(SUTTest.java:22)
	at x.SUT.delegating(SUTTest.java:19)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at x.SUT.delegating(SUTTest.java:19)
	at x.SUTTest.testDelegatingFlowsDataToStubbed(SUTTest.java:40)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)
```
Actual stack trace (the relevant bits):
```ruby
java.lang.NullPointerException
	at x.SUT.delegatee(SUTTest.java:22)
	at x.SUT.delegatee(SUTTest.java:22)
	at x.SUT.delegating(SUTTest.java:19)
	at x.SUT.delegating(SUTTest.java:19)
	at x.SUTTest.testDelegatingFlowsDataToStubbed(SUTTest.java:40)
```
Expected stack trace (the relevant bits):
```ruby
java.lang.NullPointerException
	at x.SUT.stubbed(SUTTest.java:28)
	at x.SUT.delegatee(SUTTest.java:22)
	at x.SUT.delegatee(SUTTest.java:22)
	at x.SUT.delegating(SUTTest.java:19)
	at x.SUT.delegating(SUTTest.java:19)
	at x.SUTTest.testDelegatingFlowsDataToStubbed(SUTTest.java:40)
```
Note: the duplicate stacks are ok, I don't care, but the exception source must be correctly identified. I just spent 3 hours trying to figure out why it was crashing on a line where nothing could be `null` leading to this SSCCE.

check that

 - [x] The mockito message in the stacktrace have useful information, but it didn't help
 - [x] The problematic code (if that's possible) is copied here;
       Note that some configuration are impossible to mock via Mockito
 - [x] Provide versions (mockito / jdk / os / any other relevant information)
 - [x] Provide a [Short, Self Contained, Correct (Compilable), Example](http://sscce.org) of the issue
       (same as any question on stackoverflow.com)
 - [x] Read the [contributing guide](https://github.com/mockito/mockito/blob/master/.github/CONTRIBUTING.md)

**Comment by holubec-petr** [comment]
Looks like #2238

**Comment by TWiStErRob** [comment]
Yep, duplicate of #2238; #2241 fixes the issue.
Confirmed with repro on 3.8.0 and no repro on 3.9.0 (https://github.com/mockito/mockito/commit/f5b9975e9305f596e208d840ea90c483193aadd3 included on 3.8.14 onwards)
